packages/suite/src/reducers/wallet/accountSearchReducer.ts

```diff
@@ -1,50 +1,61 @@
-import { produce } from 'immer';
+import { PayloadAction, createSlice } from '@reduxjs/toolkit';
 
 import { NetworkSymbol } from '@suite-common/wallet-config';
 import { changeNetworks, deviceActions } from '@suite-common/wallet-core';
 
-import { ACCOUNT_SEARCH } from 'src/actions/wallet/constants';
-import { Action, AppState } from 'src/types/suite';
+import { AppState } from 'src/types/suite';
 
-export interface State {
-    coinFilter: NetworkSymbol | undefined;
+export const ACCOUNT_SEARCH_PREFIX = '@suite/account-search';
+
+export type AccountSearchState = {
+    coinFilter: Array<NetworkSymbol>;
     searchString: string | undefined;
-}
+};
 
-export const initialState: State = {
-    coinFilter: undefined,
+export const accountSearchInitialState: AccountSearchState = {
+    coinFilter: [],
     searchString: undefined,
 };
 
-const accountSearchReducer = (state: State = initialState, action: Action): State =>
-    produce(state, draft => {
-        switch (action.type) {
-            case ACCOUNT_SEARCH.SET_SEARCH_STRING:
-                draft.searchString = action.payload;
-                break;
-            case ACCOUNT_SEARCH.SET_COIN_FILTER:
-                draft.coinFilter = action.payload;
-                break;
-            // reset coin filter on:
-            // 1) disabling/enabling coins
-            // 2) adding a new account is handled directly in add account modal, reacting on ACCOUNT.CREATE would cause resetting during initial accounts discovery
-            case changeNetworks.type: {
-                if (changeNetworks.match(action)) {
-                    draft.coinFilter = undefined;
-                    draft.searchString = undefined;
-                }
-                break;
+const accountSearchSlice = createSlice({
+    name: ACCOUNT_SEARCH_PREFIX,
+    initialState: accountSearchInitialState,
+    reducers: {
+        setCoinFilter(state, action: PayloadAction<Array<NetworkSymbol>>) {
+            state.coinFilter = action.payload ?? [];
+        },
+        toggleCoinFilter(state, action: PayloadAction<NetworkSymbol>) {
+            const symbol = action.payload;
+            if (!symbol) return;
+
+            if (state.coinFilter.includes(symbol)) {
+                state.coinFilter = state.coinFilter.filter(s => s !== symbol);
+            } else {
+                state.coinFilter.push(symbol);
             }
+        },
```

Comment:
What about rather using `Set` instead of `Array`? Many of these operation could be then simplified. E.g.:

```ts
toggleCoinFilter(state, action: PayloadAction<NetworkSymbol>) {
            const symbol = action.payload;
            if (!symbol) return;

            if (state.coinFilter.has(symbol) {
                 state.coinFilter.delete(symbol)
            } else {
                 state.coinFilter.add(symbol)
            }
        },
```

It's seem better suitable for this use-case – all items to be unique in the data structure.